### PR TITLE
Allow specifying no group to render a shape

### PR DESCRIPTION
**Pull request by MikeAlhayek**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs [hunk 1]
```diff
@@ -107,11 +107,16 @@ private async Task ApplyImplementationAsync(BuildShapeContext context, string di
...
         }
 
         // If the shape's group doesn't match the currently rendered one, return.
-        if (!string.IsNullOrEmpty(context.GroupId) && !_groupIds.Contains(context.GroupId, StringComparer.OrdinalIgnoreCase))
+        if (_groupIds.Count > 0 && !_groupIds.Contains(context.GroupId ?? string.Empty, StringComparer.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        if (_groupIds.Count == 0 && !string.IsNullOrEmpty(context.GroupId))
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs:119]: I think that's what `StringValues.IsNullOrEmpty` does

File: src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs [hunk 2]
```diff
@@ -312,15 +317,7 @@ public ShapeResult OnGroup(params string[] groupIds)
...
-        foreach (var groupId in groupIds)
-        {
-            if (string.IsNullOrEmpty(groupId))
-            {
-                continue;
-            }
-
-            _groupIds = StringValues.Concat(_groupIds, groupId);
-        }
+        _groupIds = StringValues.Concat(_groupIds, groupIds.Where(x => x != null).ToArray());
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs:320]: This code is allocating however

File: src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs]: updated

File: src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs]: that linq statement is not needed. fixed.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**